src/lib/PostgresMetaColumns.ts

```diff
@@ -57,75 +57,148 @@ export default class PostgresMetaColumns {
     schema?: string
   }): Promise<PostgresMetaResult<PostgresColumn>> {
     if (id) {
+      const { data, error } = await this.batchRetrieve({ ids: [id] })
+      if (data) {
+        return { data: data[0], error: null }
+      } else if (error) {
+        return { data: null, error: error }
+      }
+    }
+    if (name && table) {
+      const { data, error } = await this.batchRetrieve({ names: [name], table, schema })
+      if (data) {
+        return { data: data[0], error: null }
+      } else if (error) {
+        return { data: null, error: error }
+      }
+    }
+    return { data: null, error: { message: 'Invalid parameters on column retrieve' } }
+  }
+
+  async batchRetrieve({ ids }: { ids: string[] }): Promise<PostgresMetaResult<PostgresColumn[]>>
+  async batchRetrieve({
+    names,
+    table,
+    schema,
+  }: {
+    names: string[]
+    table: string
+    schema: string
+  }): Promise<PostgresMetaResult<PostgresColumn[]>>
+  async batchRetrieve({
+    ids,
+    names,
+    table,
+    schema = 'public',
+  }: {
+    ids?: string[]
```

Comment:
I really dislike this way of everything being optional - can we use a discriminated union type instead?

Reply:
It's a TS thing - the 2 function declaration above it means it's either `ids` or `names` + `table` + `schema`